libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp

```diff
@@ -116,13 +116,15 @@ void SLCANRouter::slcan2can_router_trampoline(void)
             _port->flush();
         }
         chSysUnlock();
-        _slcan_if.reader();
-        if (_can_tx_queue.available() && _can_if) {
+        _slcan_if.reader(false);
+        while (_can_tx_queue.available() && _can_if) {
```

Comment:
This is a curious thing to have in a patch to add UAVCAN support to SITL.  Warning signs when something changes into a while loop from an if loop - what's the bound on computation time?  Can stuff get fed into that queue faster than we can process it?

Reply:
The two changes from 'if' to 'while' are a small bug fix that allows SLCAN passthru mode to handle more than it's current limit of about 5000msgs/second, which I brought to Sids attention, but he actually fixed.

Reply:
So as @davidbuzz mentioned this is to improve throughput of CAN messages over SLCAN. Having this is not a big concern because a) this is meant for CAN Routing only, which is not to be done in-flight, but only for debugging purposes b) we do have 100us delay to allow lower priority threads to be invoked.

I know I have a bad habit of adding small slightly off-topic changes in big PRs that ought to be separate. I will try to be vigilant in the future. If you want I can pull this out into a separate PR.